libevmasm/RuleList.h

```diff
@@ -581,9 +581,20 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 
 	rules.push_back({
 		Builtins::BYTE(A, Builtins::SHR(B, X)),
-		[=]() -> Pattern { return A.d() < B.d() / 8 ? Word(0) : Builtins::BYTE(A.d() - B.d() / 8, X); },
+		[=]() -> Pattern { return Word(0); },
+		true,
+		[=] {
+			return B.d() % 8 == 0 && A.d() < Pattern::WordSize / 8 && B.d() <= Pattern::WordSize && A.d() < B.d() / 8;
```

Comment:
In a follow-up pull request, we could actually drop the `%8==0` requirement here.

Reply:
True - then we should drop it from ``test/formal/combine_byte_shr_2.py`` as well, though. Rather unfortunate - I did split up the proofs back then and even thought about suggesting splitting up the rules, but saw no reason for it, since I had already stopped thinking about the flag by then...